ft_badchannel.m

```diff
@@ -0,0 +1,204 @@
+    ft_info('median(range) = %g\n', median(chanrange));
+    ft_info('mad(range)    = %g\n', mad(chanrange));
+    
+    % find channels above the threshold
+    bad = chanrange>cfg.threshold;
+    
+  case 'std'
+    % compute the standard deviation for each channel
+    chanstd = nan(size(lab));
+    for i=1:nchan
+      chanstd(i) = std(dat(i,:));
+    end
+    
+    % give some feedback
+    ft_info('min(range)    = %g\n', min(chanstd));
+    ft_info('max(range)    = %g\n', max(chanstd));
+    ft_info('mean(range)   = %g\n', mean(chanstd));
+    ft_info('median(range) = %g\n', median(chanstd));
+    ft_info('mad(range)    = %g\n', mad(chanstd));
+    
+    % find channels above the threshold
+    bad = chanstd>cfg.threshold;
+    
+  case 'neighstdratio'
+    % compute the standard deviation for each channel
+    chanstd = nan(size(lab));
+    for i=1:nchan
+      chanstd(i) = std(dat(i,:));
+    end
+    
+    chanstdratio = nan(length(lab), length(lab));
+    for i=1:nchan
+      selthis = find(strcmp({cfg.neighbours.label}, lab{i}));
+      selnb   = find(ismember(lab, cfg.neighbours(selthis).neighblabel));
+      
+      labnb = sprintf('%s, ', lab{selnb});
+      labnb = labnb(1:end-2);
+      ft_debug('channel %s has neighbours %s\n', lab{i}, labnb);
+      
+      for j=selnb(:)'
+        chanstdratio(i,j) = abs(chanstd(i)-chanstd(j)) / chanstd(j);
+      end
+    end
+    
+    % give some feedback
+    tmp = chanstdratio(~isnan(chanstdratio(:)));
+    ft_info('min(range)    = %g\n', min(tmp));
+    ft_info('max(range)    = %g\n', max(tmp));
+    ft_info('mean(range)   = %g\n', mean(tmp));
+    ft_info('median(range) = %g\n', median(tmp));
+    ft_info('mad(range)    = %g\n', mad(tmp));
+    figure; histogram(tmp, round(length(tmp)/10));
+    figure; imagesc(chanstdratio);
+    
+    % find channels with a median neighbour ABOVE the threshold
+    bad = nanmedian(chanstdratio,2)>cfg.threshold;
+    
+  case 'neighcorrel'
+    channeighcorrel = nan(length(lab), length(lab));
+    for i=1:nchan
+      selthis = find(strcmp({cfg.neighbours.label}, lab{i}));
+      selnb   = find(ismember(lab, cfg.neighbours(selthis).neighblabel));
+      
+      labnb = sprintf('%s, ', lab{selnb});
+      labnb = labnb(1:end-2);
+      ft_debug('channel %s has neighbours %s\n', lab{i}, labnb);
+      
+      for j=selnb(:)'
+        channeighcorrel(i,j) = corr(dat(i,:)', dat(j,:)');
+      end
+    end
+    
+    % give some feedback
+    tmp = channeighcorrel(~isnan(channeighcorrel(:)));
+    ft_info('min(range)    = %g\n', min(tmp));
+    ft_info('max(range)    = %g\n', max(tmp));
+    ft_info('mean(range)   = %g\n', mean(tmp));
+    ft_info('median(range) = %g\n', median(tmp));
+    ft_info('mad(range)    = %g\n', mad(tmp));
+    figure; histogram(tmp, round(length(tmp)/10));
+    figure; imagesc(channeighcorrel);
+    
+    % find channels with a median neighbour value BELOW the threshold
+    bad = nanmedian(channeighcorrel,2)<cfg.threshold;
```

Comment:
each channel has a number of neighbours with which the correlation can be low. 

Imagine a single bad channel surrounded by good chanels: all good channels would have a low correlation with the bad channel. To prevent the good ones to be marked as bad, the good ones need more neighbors with which they would have a high correlation. So the good ones correlate only poorly with a (single) bad one, and correlate well with other neighbours.

The bad one correlates poorly with _all_ its neighbors.